README.md

```diff
@@ -1,101 +1,101 @@
-
-    $ gem install chef
+If you are not using bundler, you can install the gem manually from Rubygems:
 
-This plugin is distributed as a Ruby Gem. To install it, run:
-
-    $ gem install knife-ec2
+```bash
+$ gem install knife-ec2
+```
 
 Depending on your system's configuration, you may need to run this command with root privileges.
 
+## Configuration
+In order to communicate with the Amazon's EC2 API you will need to pass Knife your AWS Access Key, Secret Access Key, and if using STS your session token. This can be done in several ways:
 
-Configuration
--------------
-In order to communicate with the Amazon's EC2 API you will have to tell Knife about your AWS Access Key and Secret Access Key. The easiest way to accomplish this is to create some entries in your `knife.rb` file:
+### Knife.rb Configuration
+The easiest way to configure your Amazon EC2 credentials for knife-ec2 is to specify them in your your `knife.rb` file:
 
 ```ruby
 knife[:aws_access_key_id] = "Your AWS Access Key ID"
 knife[:aws_secret_access_key] = "Your AWS Secret Access Key"
 ```
 
-If your `knife.rb` file will be checked into a SCM system (ie readable by others) you may want to read the values from environment variables:
+Additionally if using AWS STS:
+
+```ruby
+knife[:aws_session_token] = "Your AWS Session Token"
+```
+
+Note: If your `knife.rb` file will be checked into a source control management system, or is otherwise accessible by others, you may want to use one of the other configuration methods to avoid exposing your credentials.
+
+### Environmental Variables
+Knife-ec2 can also read your credentials from shell environmental variables. Export `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN` variables in your shell then add the following configuration to your `knife.rb` file:
 
 ```ruby
 knife[:aws_access_key_id] = ENV['AWS_ACCESS_KEY_ID']
 knife[:aws_secret_access_key] = ENV['AWS_SECRET_ACCESS_KEY']
-# Optional if you're using Amazon's STS
-knife[:aws_session_token] = ENV['AWS_SESSION_TOKEN']
 ```
 
-You also have the option of passing your AWS API Key/Secret into the individual knife subcommands using the `-A` (or `--aws-access-key-id`) `-K` (or `--aws-secret-access-key`) command options
+Additionally if using AWS STS:
 
-```bash
-# provision a new m1.small Ubuntu 10.04 webserver
-$ knife ec2 server create -r 'role[webserver]' -I ami-7000f019 -f m1.small -A 'Your AWS Access Key ID' -K "Your AWS Secret Access Key"
+```ruby
+knife[:aws_session_token] = ENV['AWS_SESSION_TOKEN']
 ```
 
-If you are working with Amazon's command line tools, there is a good chance
-you already have a file with these keys somewhere in this format:
+### CLI Arguments
+You also have the option of passing your AWS API Key/Secret into the individual knife subcommands using the `-A` (or `--aws-access-key-id`) `-K` (or `--aws-secret-access-key`) command options
```

Comment:
looks like -A gets clobbered by -A for --forward-agent. Let's not list the short codes at all here, i.e. leave off both -A and -K. This is a silly way to pass credentials anyway.